test/routs/testSoldierRouts.js

```diff
@@ -71,12 +79,194 @@ describe("Soldiers tests", function () {
         "stam": "lol",
         "limitations": ["yeshiva"]
       }));
-      Http.onreadystatechange = (e) => {
-        if (Http.readyState == 4 && Http.status == 200) {
-          expect(Http.responseText).to.eql("One or more fields is invalid");
-          done();
+      Http.onreadystatechange = (connetErr) => {
+        if (connetErr) {
+          done(connetErr);
+        } else {
+          if (Http.readyState == 4 && Http.status == 200) {
+            expect(Http.responseText).to.eql("One or more fields is invalid");
+            done();
+          }
+        }
+      }
+    })
+  });
+
+  describe("Soldiers get test", function () {
+
+    it("Should be able to return correct respone when the soldier path doesn't contain id", function (done) {
+      const Http = new XMLHttpRequest();
+      Http.open("GET", serverUrl);
+      Http.send();
+      Http.onreadystatechange = (connetErr) => {
+        if (connetErr) {
+          done(connetErr);
+        } else {
+          if (Http.readyState == 4 && Http.status == 200) {
+            MongoClient.connect(DBurl, function (err, client) {
+              assert.equal(null, err);
+              const db = client.db(dbName);
+              const collection = db.collection('Soldiers')
+              collection.find({}).toArray(function (err, docs) {
+                if (err) {
+                  done(err);
+                } else {
+                  if (docs.length === 1) {
+                    docs = docs[0];
+                  }
+                  client.close();
+                  expect(Http.responseText).to.eql(JSON.stringify(docs));
+                  done();
+                }
+              });
+            });
+          }
         }
       }
     })
+
+    it("Should be able to return correct respone when the soldier path contain specific soldier information", function (done) {
```

Comment:
You can remove the "be able to" part as it only increases the length of the already  long test name.
And even more important: what is the correct response? Keep it short or make it informative.